src/main/java/org/assertj/core/error/ShouldHaveSameFileSystem.java

```diff
@@ -0,0 +1,47 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ *
+ * Copyright 2012-2022 the original author or authors.
+ */
+package org.assertj.core.error;
+
+import java.nio.file.FileSystem;
+import java.nio.file.Path;
+import org.assertj.core.util.VisibleForTesting;
+
+/**
+ * Creates an error message indicating that an assertion that verifies the {@link FileSystem} for
+ * a given path does not match a given file system.
+ *
+ * @author Ashley Scopes
+ */
+public class ShouldHaveSameFileSystem extends BasicErrorMessageFactory {
```

Comment:
extract `ShouldHaveFileSystem` from `ShouldHaveSameFileSystem` and only keep `shouldHaveSameFileSystemAs` here

Reply:
I've split it up into two classes now. ShouldHaveSameFileSystem and ShouldHaveSameFileSystemAsPath. I assume that is what you mean by the naming?